Drop Pulp 2 Smart Proxy status code

ekohl opened the pull request:
#### What are the changes introduced in this pull request?

It drops the Pulp 2 Smart Proxy status extensions.

#### Considerations taken when implementing this change?

These were likely not triggered anymore because the Pulp 2 features were never present.

#### What are the testing steps for this pull request?

Go to the Smart Proxies overview and for every Smart Proxy it should still render well. I forgot if there's a Pulp 3 tab today, but if there is, it should still be present after.

ianballou commented:
Guessing we missed this since it didn't show up in triage via redmine, mind adding one so we can get it on our job board?

ianballou reviewed (changes requested):
engine.rb still has `::SmartProxiesController.include Katello::Concerns::SmartProxiesControllerExtensions` in it, removing that should fix the tests.

ekohl commented:
Rebased and removed require statements.

ekohl commented on app/controllers/katello/concerns/smart_proxies_controller_extensions.rb:
```diff
@@ -1,68 +0,0 @@
-      module Overrides
-        def show
-          @task_search_url = main_app.foreman_tasks_tasks_path(:search => "resource_id = #{@smart_proxy.id} AND resource_type = #{@smart_proxy.class}")
-          render 'foreman/smart_proxies/show', :layout => 'katello/layouts/foreman_with_bastion'
```
I'm not 100% sure if this also needs to be reflected in any templates.

ianballou reviewed (commented):
Looks like it's breaking the smart proxy details pages:
```
2024-07-16T20:18:30 [F|app|3fb86828]   
 3fb86828 | ActionView::Template::Error (undefined method `map' for nil:NilClass):
 3fb86828 |     1: <%= render :partial => 'disk_usage', :collection => disks(@storage), :as => :disk %>
 3fb86828 |   
 3fb86828 | /home/vagrant/katello/app/helpers/katello/concerns/smart_proxy_helper_extensions.rb:5:in `disks'
 3fb86828 | /home/vagrant/katello/app/views/smart_proxies/pulp_storage.html.erb:1
 3fb86828 | app/controllers/concerns/foreman/controller/timezone.rb:10:in `set_timezone'
 3fb86828 | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 3fb86828 | app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
 3fb86828 | /home/vagrant/katello/lib/katello/middleware/organization_created_enforcer.rb:18:in `call'
 3fb86828 | /home/vagrant/katello/lib/katello/middleware/event_daemon.rb:10:in `call'
 3fb86828 | lib/foreman/middleware/libvirt_connection_cleaner.rb:9:in `call'
 3fb86828 | lib/foreman/middleware/telemetry.rb:10:in `call'
 3fb86828 | lib/foreman/middleware/logging_context_session.rb:22:in `call'
 3fb86828 | lib/foreman/middleware/logging_context_request.rb:11:in `call'
 3fb86828 | /home/vagrant/katello/lib/katello/prevent_json_parsing.rb:12:in `call'
2024-07-16T20:18:32 [I|app|3c441f02] Processing by Katello::Api::V2::CapsuleContentController#sync_status as JSON
```

ianballou commented:
I think the Pulp and Pulp Node bits can also be removed from: https://github.com/katello/katello/blob/master/lib/katello/plugin.rb#L313

ianballou commented on app/controllers/katello/concerns/smart_proxies_controller_extensions.rb:
```diff
@@ -1,68 +0,0 @@
-        before_action :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
-
-        def pulp_storage
-          @storage = @smart_proxy.pulp_disk_usage
```
I'm not sure how much of this file is still used, but this bit definitely is still used to show the Pulp 3 storage amount. `@storage` being `nil` in the error I pasted earlier makes sense since this file was deleted.

ekohl committed:
Fixes #37388 - Drop Pulp 2 related code

The PULP_FEATURE and PULP_NODE_FEATURE features were for Pulp 2. This is
never present in a current Katello installation and can be dropped.

It also replaces the deprecated default_capsule code, which is replaced
by pulp_primary.

In the template the code to handle primaries is disabled because it's
only shown for mirrors. A mirror is by definition not a primary, so it
can never trigger.

ekohl commented on app/assets/javascripts/katello/containers/container.js:
```diff
@@ -55,6 +55,7 @@ KT.container = (function(){
     getTag = function() {
         return $('#tag_id').val();
     },
+    // TODO: This only returns Pulp 2 proxies, which no longer exist
```
I can't judge if this whole page is now redundant (was it the Pulp 2 container feature?) or if it needs to be updated to Pulp 3 proxies.

Given it mentions the `docker` content type, I have a feeling it's Pulp 2.

ekohl replied:
https://github.com/Katello/katello/pull/11078

ekohl commented on test/support/capsule_support.rb:
```diff
@@ -1,10 +1,9 @@
 module Support
   module CapsuleSupport
```
I think this should be done in factories as traits, but I considered it out of scope here.

ekohl commented on app/models/katello/concerns/smart_proxy_extensions.rb:
```diff
@@ -99,24 +91,8 @@ def self.pulp_primary!
+          pulpcore_proxies_with_environment(environment).try(:uniq)
         end
 
         def self.pulpcore_proxies_with_environment(environment)
```
I thought about optimizing this to:
```ruby
unscoped.select { |p| p.pulp_mirror? }.joins(:capsule_lifecycle_environments).
  where(katello_capsule_lifecycle_environments: { lifecycle_environment_id: environment.id }).distinct
```

ianballou replied:
This would be fine, I like avoiding excess queries where possible.

ekohl commented on app/models/katello/concerns/smart_proxy_extensions.rb:
```diff
@@ -99,24 +91,8 @@ def self.pulp_primary!
-          pulp_primary!
-        end
-
         def self.with_environment(environment, include_default = false)
```
```suggestion
        def self.with_environment(environment)
```

ianballou replied:
This should be fine, I don't see any uses of `include_default = false` with `with_environment()`.

ianballou reviewed (commented):
Something is still broken on the smart proxies page, looks like `storage` is still ending up `nil`.

```
 48a2c081 | ActionView::Template::Error (undefined method `map' for nil:NilClass):
 48a2c081 |     1: <%= render :partial => 'disk_usage', :collection => disks(@storage), :as => :disk %>
 48a2c081 |   
 48a2c081 | /home/vagrant/katello/app/helpers/katello/concerns/smart_proxy_helper_extensions.rb:5:in `disks'
 48a2c081 | /home/vagrant/katello/app/views/smart_proxies/pulp_storage.html.erb:1
 48a2c081 | app/controllers/concerns/foreman/controller/timezone.rb:10:in `set_timezone'
 48a2c081 | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 48a2c081 | app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
 48a2c081 | /home/vagrant/katello/lib/katello/middleware/organization_created_enforcer.rb:18:in `call'
 48a2c081 | /home/vagrant/katello/lib/katello/middleware/event_daemon.rb:10:in `call'
 48a2c081 | lib/foreman/middleware/libvirt_connection_cleaner.rb:9:in `call'
 48a2c081 | lib/foreman/middleware/telemetry.rb:10:in `call'
 48a2c081 | lib/foreman/middleware/logging_context_session.rb:22:in `call'
 48a2c081 | lib/foreman/middleware/logging_context_request.rb:11:in `call'
 48a2c081 | /home/vagrant/katello/lib/katello/prevent_json_parsing.rb:12:in `call'
```

ianballou reviewed (commented):
I haven't figured out why, but this PR is stopping find_resource from working. I have to manually add `find_resource` to controller methods for them to work in both Foreman and Katello. For example, there are nil errors in smart_proxies_controller#modules_card for `@smart_proxy.statuses`

ekohl commented:
@ianballou would you be willing to take this over? I don't have the time to focus on this, but I still think cleaning up unused code is something that should be done.

ianballou commented:
> @ianballou would you be willing to take this over? I don't have the time to focus on this, but I still think cleaning up unused code is something that should be done.

Sure, I can take it over :+1:

ianballou committed:
Refs #37388 - remove use of find_resource that broke callback chain

ianballou commented:
Closing because the rest of the work is in https://github.com/Katello/katello/pull/11811
